### ISSUE TITLE
OSK doesn't work anymore if you connect a bluetooth mouse

### ISSUE THREAD

**Issue by pixelr0** [ubports/ubuntu-touch#367]
i see this on M10/ota2. connect a bluetooth mouse and then try to login or touch any text input field. the OSK doesn't show up anymore... so you can't even login

**Comment by TronFortyTwo** [comment]
Yes. When you connect an external mouse the device supposes you're using an external keyboard too and prevent OSK to appear. I reproduce this bug on bacon and vegetahd too so I think is device agnostic. I tested it with a mouse connected with usb otg and a usb to ps2 adapter, so I think it happens with any mouse.

**Comment by Flohack74** [comment]
This is questionable if this is a bug - whats the purpose of using an external mouse only, but no keyboard?

**Comment by pixelr0** [comment]
but there's more, the OSK doesn't work even after you remove/turn off the mouse. so to see the OSK again you need to reboot

**Comment by Flohack74** [comment]
Ok damn, then this is a big bug ^^

**Comment by jonnius** [comment]
If it is about the OSK not showing after removing the mouse, it is a duplicate of #351. Otherwise I would not consider this one a bug. Or at least it would be a wontfix and could be closed imho.

**Comment by luksus42** [comment]
> whats the purpose of using an external mouse only, but no keyboard?
mh, I would want to watch a movie on a big screen on my sofa.
There I have no mouse and no keybord, but a steamcontroller.
Let's say I want to find a movie in my plex-server: most of the time a mouse will be enough. But sometimes I need to type a short search-term, a login and password (netflix), or similar...

Either an OSK on the phone OR on the external display would be helpful.

**Comment by mateosalta** [comment]
yes, I have a few mouse only peripherals, and also on the tablet with using a mouse only via otg - I would like the osk to appear (or the option to still have it appear in settings) while using a single mouse peripheral, or  presentation remote(like a hand held mouse). Having to disconnect the mouse to type something is not very user friendly.

**Comment by kugiigi** [comment]
I think this happens when connecting a bluetooth mouse with extra buttons such as page up and page down. They probably has keyboard profile as well so UT detects it as both mouse and keyboard and thus hiding the OSK. As mentioned though, the OSK doesn't reappear after disconnecting so it's definitely a bug. BTW, I have a BT mouse that doesn't have this issue.

**Comment by luksus42** [comment]
Oh, I thought this was solved already...?!
Because some weeks ago, I tried this again and an OSK did appear, which didn't when I tried it the last time.

**Comment by kugiigi** [comment]
> Oh, I thought this was solved already...?!
> Because some weeks ago, I tried this again and an OSK did appear, which didn't when I tried it the last time.

Perhaps you use a different mouse? I just encountered this a few days ago. They were logitech multi device mouse. My cheap Rapoo BT mouse works fine though.

**Comment by kugiigi** [comment]
There's a new option in the edge channel to enable OSK even when an external keyboard is connected.
This can be used as a workaround once it lands in stable.
But the bug still persists, an external keyboard is still detected when connecting specific types of BT mouse and even after disconnecting. I guess when connected, the keyboard profile from the mouse is considered but not when disconnecting.

**Comment by dobey** [comment]
> I guess when connected, the keyboard profile from the mouse is considered but not when disconnecting.

The property should be updated when disconnecting. If it's not, perhaps there's a deeper bug somewhere.

**Comment by mateosalta** [comment]
maybe a setting to click in settings, to say "yes I'm only using a mouse" - 'Show osk always' - there are many times I can bring a small mouse with, but packing a keyboard along with the tablet is too much.

**Comment by dobey** [comment]
> maybe a setting to click in settings, to say "yes I'm only using a mouse" - 'Show osk always' - there are many times I can bring a small mouse with, but packing a keyboard along with the tablet is too much.

There already is a "Always enable OSK" option in system-settings.

**Comment by luksus42** [comment]
@dobey : I can't find such an option. Where exactly is it?

**Comment by kugiigi** [comment]
It is in the Language & Text page. But it only shows when an external keyboard is connected. Please note that it's in the edge channel only.
Also the keyboard icon in the login/lock screen in the new Unity 8 also has this function.

**Comment by luksus42** [comment]
Ok, thanks for the explanation.

**Comment by dobey** [comment]
> It is in the Language & Text page. But it only shows when an external keyboard is connected. Please note that it's in the edge channel only.
> Also the keyboard icon in the login/lock screen in the new Unity 8 also has this function.

It's also in system-settings in stable. If it's not showing there though, I have no idea why. It should.
